src/main/kotlin/org/cascadebot/cascadebot/data/objects/GuildSettingsModeration.kt

```diff
@@ -0,0 +1,227 @@
+            this.webhookId = webhookId
+            this.webhookToken = webhookToken
+            this.id = id
+            buildWebhookClient()
+        }
+
+        fun buildWebhookClient() {
+            webhookClient = WebhookClientBuilder(webhookId, webhookToken).build()
+        }
+
+        fun getId(): Int {
+            return id
+        }
+
+        fun getEvents(): List<ModlogEvent> {
+            return ArrayList(events)
+        }
+
+        fun addEvent(event: ModlogEvent): Boolean {
+            return events.add(event)
+        }
+
+        fun removeEvent(event: ModlogEvent): Boolean {
+            return events.remove(event)
+        }
+
+        fun getWebhookId() : Long {
+            return webhookId
+        }
+
+        fun setNewWebhook(webhookId: Long, webhookToken: String) {
+            this.webhookId = webhookId
+            this.webhookToken = webhookToken
+            buildWebhookClient()
+        }
+
+        fun sendEvent(guildData: GuildData, modlogEventStore: ModlogEventStore) {
+            val webhookEmbedBuilder = WebhookEmbedBuilder()
+            webhookEmbedBuilder.setTitle(EmbedTitle(i18n(guildData.locale, "enums.moldogevent." + modlogEventStore.trigger.name.toLowerCase() + ".display"), null))
+            val affected: ISnowflake = modlogEventStore.affected;
+            var affectedType = ""
+            val affectedStr = when (affected) {
+
+                is User -> {
+                    affectedType = "User";
+                    affected.name + " (" + affected.id + ")"
+                }
+                is Role -> {
+                    affectedType = "Role"
+                    affected.name + " (" + affected.id + ")"
+                }
+                is Emote -> {
+                    affectedType = "Emote"
+                    affected.name
+                }
+                is Guild -> {
+                    affectedType = "Guild"
+                    affected.name
+                }
+                is GuildChannel -> {
+                    affectedType = "Channel"
+                    affected.name
+                }
+                else -> null
+            }
+            if (affectedStr != null) {
+                webhookEmbedBuilder.addField(EmbedField(true, "Affected $affectedType", affectedStr))
+            }
+            for (embedField in modlogEventStore.extraInfo) {
+                webhookEmbedBuilder.addField(embedField.getLocalizedEmbedField(guildData.locale))
+            }
+            webhookEmbedBuilder.setColor(modlogEventStore.trigger.messageType.color.rgb)
+            if (modlogEventStore.responsible != null) {
+                webhookEmbedBuilder.setFooter(WebhookEmbed.EmbedFooter(modlogEventStore.responsible!!.name + " (" + modlogEventStore.responsible!!.id + ")", null))
+            }
+            try {
+                webhookClient?.send(webhookEmbedBuilder.build())
+            } catch (ignored: HttpException) {
+                // TODO not ignore
```

Comment:
Yeah we need to log these as WARn

Reply:
We need to talk about how to actually handle this. Logging this as a warn will create way too much spam considering this can easily happen if they delete eater the channel or the webhook.

Reply:
We should probably disable all events in the channel that caused this, granted we should also talk about it like I originally said lol.